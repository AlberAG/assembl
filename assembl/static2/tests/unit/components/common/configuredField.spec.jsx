import React from 'react';
import renderer from 'react-test-renderer';

import ConfiguredField from '../../../../js/app/components/common/configuredField';

describe('ConfiguredField component', () => {
  it('should match ConfiguredField snapshot (TEXT)', () => {
    const handleValueChangeSpy = jest.fn();
    const props = {
      configurableField: {
        fieldType: 'TEXT',
        id: 'my-field',
        required: true,
        title: 'My field',
        __typename: 'TextField'
      },
      handleValueChange: handleValueChangeSpy,
      value: 'foobar'
    };
    const rendered = renderer.create(<ConfiguredField {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should match ConfiguredField snapshot (PASSWORD)', () => {
    const handleValueChangeSpy = jest.fn();
    const props = {
      configurableField: {
        fieldType: 'PASSWORD',
        id: 'my-field',
        required: true,
        title: 'My password field',
        __typename: 'TextField'
      },
      handleValueChange: handleValueChangeSpy,
      value: 'foobar'
    };
    const rendered = renderer.create(<ConfiguredField {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should match ConfiguredField snapshot (EMAIL)', () => {
    const handleValueChangeSpy = jest.fn();
    const props = {
      configurableField: {
        fieldType: 'EMAIL',
        id: 'my-field',
        required: false,
        title: 'My email field',
        __typename: 'TextField'
      },
      handleValueChange: handleValueChangeSpy,
      value: 'foobar'
    };
    const rendered = renderer.create(<ConfiguredField {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});