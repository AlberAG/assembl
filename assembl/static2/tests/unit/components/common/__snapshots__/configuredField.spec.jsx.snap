// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfiguredField component should match ConfiguredField snapshot (EMAIL) 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="my-field"
  >
    My email field
  </label>
  <input
    className="form-control"
    disabled={true}
    id="my-field"
    name="my-field"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="My email field"
    type="text"
    value="foobar"
  />
</div>
`;

exports[`ConfiguredField component should match ConfiguredField snapshot (PASSWORD) 1`] = `null`;

exports[`ConfiguredField component should match ConfiguredField snapshot (TEXT) 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="my-field"
  >
    My field*
  </label>
  <input
    className="form-control"
    disabled={false}
    id="my-field"
    name="my-field"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="My field*"
    type="text"
    value="foobar"
  />
</div>
`;
